android/authentication.md

```diff
@@ -991,3 +1064,459 @@ public class AuthenticatorActivity extends Activity {
+{: .callout .callout--info}
+
+To learn [what's Authorized scope](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-social-idp.html#cognito-user-pools-social-idp-step-2)
+{: .callout .callout--info}
+
+Note: your user pool domain is something like: `domain_prefix.auth.us-east-1.amazoncognito.com`
+{: .callout .callout--info}
+
+- To retrieve user attributes from your identity provider, go to *Federation* > *Attribute mapping*. Here, you can map Federation Provider attributes to corresponding User pool attributes. More info about [Attribute Mapping](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-specifying-attribute-mapping.html).
+
+If the attribute, for example *email*, is a required field in your Cognito User Pool settings, please make sure that you have selected *email* in your Authorized Scopes, and you have mapped it correctly to your User Pool attributes.
+{: .callout .callout-info}
+
+To setup App Client:
+- Go to [Amazon Cognito Console](https://aws.amazon.com/cognito/).
+- Click *User Pools* on the top menu to select a User Pool or create a new one.
+- Click *App integration*  and *App client settings* on the left menu.
+- Select *Enabled Identity Providers* and enter *Callback URL(s)* and *Sign out URL(s)* fields.
+
+In *Callback URL(s)*, for both *Signin URL(s)* and *Signout URL(s)*, enter `myapp://`.
+{: .callout .callout--info}
+
+- Under the *OAuth 2.0* section, Choose OAuth Flow and OAuth scopes. [To learn more about flows and scopes.](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-app-idp-settings.html)
+- Select an OAuth Flow.
+
+By using *Authorization code grant* the callback URL will contain a code after login. The code will be used to exchange for tokens from Cognito with the TOKEN Endpoint.
+{: .callout .callout--info}
+
+*Authorization code grant* is the recommended choice for security reasons.
+{: .callout .callout--info}
+
+- Choose item(s) from *OAuth Scopes*.
+
+Note: `openid` is required for `phone`, `email` or `profile`. Also `openid` is required to get the id token from the Cognito authorization server.
+{: .callout .callout--info}
+
+- Click 'Save Changes'.
+
+#### Setup Amazon Cognito Hosted UI in Android App
+
+1. To configure your application for hosted UI, you need to use *HostedUI* options. Update your `awsconfiguration.json` file to add a new configuration for `Auth`. The configuration should look like this:
+
+    ```json
+    {
+        "IdentityManager": {
+            ...
+        },
+        "CredentialsProvider": {
+            ...
+        },
+        "CognitoUserPool": {
+            ...
+        },
+        "Auth": {
+            "Default": {
+                "OAuth": {
+                    "WebDomain": "https://YOUR_AUTH_DOMAIN.auth.us-west-2.amazoncognito.com",
```

Comment:
Can we use the same keys as the JS side: https://aws-amplify.github.io/docs/js/authentication#configuring-the-hosted-ui

Reply:
The spec that this file is currently following is camel-case with first letter capitalization. This followed the spec defined by an already launched iOS Info.plist and we had no reason to change this. We should, if necessary, make this change in a different version of this file.